UserManagement/accounts/repositories/implementations.py

```diff
@@ -11,15 +12,41 @@ def get_by_id(self, user_id):
             return UserProfile.objects.get(id=user_id)
         except ObjectDoesNotExist:
             return None
-
+        except IntegrityError as e:
+            # Handle specific database integrity issues
+            # Placeholder for future logging
+            # Log the error message: print(f"Integrity error: {e}")
+            return None
+        except Exception as e:
+            # Handle other unexpected exceptions
+            # Placeholder for future logging
+            # Log the error message: print(f"Unexpected error: {e}")
+            return None
+        
     def create(self, **kwargs):
         user_profile = UserProfile.objects.create(**kwargs)
         # Handle additional logic specific to user profile creation if necessary
         return user_profile
 
-    def update(self, user_id, **kwargs):
-        UserProfile.objects.filter(id=user_id).update(**kwargs)
-        return self.get_by_id(user_id)
+    def update(self, user_id, **user_data):
+        try:
+            user = UserProfile.objects.get(id=user_id)
+            for field, value in user_data.items():
+                setattr(user, field, value)
+            user.save()
+            return user
+        except ObjectDoesNotExist:
+            return None
+        except IntegrityError as e:
```

Comment:
Do we want to do someting if we catch any erros?

Reply:
i'm still not sure what we need to do